spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/Spark3Util.java

```diff
@@ -781,10 +801,29 @@ public static TableIdentifier identifierToTableIdentifier(Identifier identifier)
             Object value = CatalystTypeConverters.convertToScala(catalystValue, field.dataType());
             values.put(field.name(), String.valueOf(value));
           });
-          return new SparkPartition(values, partition.path().toString(), format);
+          FileStatus fileStatus =
+              scala.collection.JavaConverters.seqAsJavaListConverter(partition.files()).asJava().get(0);
+
+          return new SparkPartition(values, fileStatus.getPath().getParent().toString(), format);
         }).collect(Collectors.toList());
   }
 
+  private static List<org.apache.spark.sql.catalyst.expressions.Expression> getPartitionFilterExpressions(
+      SparkSession spark, String tableName, Map<String, String> partitionFilter) {
+    List<org.apache.spark.sql.catalyst.expressions.Expression> filterExpressions = Lists.newArrayList();
+    for (Map.Entry<String, String> entry : partitionFilter.entrySet()) {
+      String filter = entry.getKey() + " = '" + entry.getValue() + "'";
```

Comment:
why do we have quotes ('') for the value? 
Example: If `id` is an integer column, then we need `id = 3` in the query instead of `id ='3'`

have we tested both string and non string partition columns ?

I was also thinking instead of map, can we expose the `where` clause in the call procedure (similar to rewrite_data files), so user can give filters other than equals also?

Reply:
> I was also thinking instead of map, can we expose the where clause in the call procedure (similar to rewrite_data files), so user can give filters other than equals also?

But it will become breaking API changes I guess. Let's see what others think on this also.

Reply:
If the filter is on String column, e.g. `dept` column with value `hr`, we want the filter to be `dept = 'hr'`.

For non-String columns, such as `id = 3`, the filter with `id = '3'` still works ok because Spark will have `Literal` with String value to begin with, and then cast to `Literal` with Int value after it has the column type.

`TestAddFilesProcedure` has both String and int type partition columns so these two are tested. We probably should test other types e.g. Timestamp just to make sure.